project-base/webpack.config.js

```diff
@@ -47,16 +47,146 @@ Encore
+        { from: 'web/bundles/fpjsformvalidator', to: '../../assets/js/bundles/fpjsformvalidator', force: true },
+        { from: 'web/bundles/shopsysframework', to: '../../web/assets/admin/images', force: true }
     ]))
 ;
 
+// Frontend Config
+const WebfontsGenerator = require('webfonts-generator');
+const StylelintPlugin = require('stylelint-webpack-plugin');
+var requireContext = require('require-context');
+yaml = require('js-yaml');
+fs = require('fs');
+
+SVGO = require('svgo');
+const svgo = new SVGO();
+
+const svgFilesFrontend = requireContext('../../src/Resources/svg/front', false, '.svg');
+var svgFilesFrontendPath = [];
+svgFilesFrontend.keys().forEach((filePathFrontend) => {
+    var newFileFrontendPath = './src/Resources/svg/front/' + filePathFrontend;
+    svgFilesFrontendPath.push(newFileFrontendPath);
+
+    fs.readFile(newFileFrontendPath, 'utf8', function (err, data) {
+        if (err) {
+            throw err;
+        }
+        svgo.optimize(data, { path: newFileFrontendPath }).then(function (result) {
+            fs.writeFile(newFileFrontendPath, result.data, function (err) {
+                if (err) {
+                    throw err;
+                }
+                console.log('Frontend SVG icon ' + newFileFrontendPath + ' optimized');
+            });
+        });
+    });
+});
+
+const svgFilesAdmin = requireContext('../../../packages/framework/src/Resources/svg/admin', false, '.svg');
+var svgFilesAdminPath = [];
+svgFilesAdmin.keys().forEach((filePathAdmin) => {
+    var newFileAdminPath = '../packages/framework/src/Resources/svg/admin/' + filePathAdmin;
+    svgFilesAdminPath.push(newFileAdminPath);
+
+    fs.readFile(newFileAdminPath, 'utf8', function (err, data) {
+        if (err) {
+            throw err;
+        }
+        svgo.optimize(data, { path: newFileAdminPath }).then(function (result) {
+            fs.writeFile(newFileAdminPath, result.data, function (err) {
+                if (err) {
+                    throw err;
+                }
+                console.log('Admin SVG icon ' + newFileAdminPath + ' optimized');
+            });
+        });
+    });
+});
+
+const domainFile = './config/domains.yml';
+var domains = yaml.safeLoad(fs.readFileSync(domainFile, 'utf8'));
+
+domains.domains.forEach((domain) => {
+    if (!domain.styles_directory) {
+        domain.styles_directory = 'common';
+    }
+    Encore
+        .addEntry('frontend-style-' + domain.styles_directory, './src/Resources/styles/front/' + domain.styles_directory + '/main.less')
+        .addEntry('frontend-print-style-' + domain.styles_directory, './src/Resources/styles/front/' + domain.styles_directory + '/print/main.less');
+});
+
+Encore
+    .addEntry('admin-style', '../packages/framework/src/Resources/styles/admin/main.less')
+    .addPlugin(
+        new StylelintPlugin({
+            configFile: '.stylelintrc',
+            files: 'src/**/*.less'
+        })
+    )
+    .addLoader(
+        new WebfontsGenerator({
```

Comment:
This is another DRY violation. This configuration is similar to WebfontsGenerator for admin. Maybe I recommend creating a new function that will return the configuration for WebfontsGenerator

Reply:
I agree...

Reply:
This part was refactored and used as local function from assets/js/command file